DIPs/DIP1xxx-sw.md

```diff
@@ -0,0 +1,536 @@
+```
+
+Also the overloads could be reduced to a single one:
+
+```
+struct SpawnParams
+{
+    File stdin = std.stdio.stdin;
+    File stdout = std.stdio.stdout;
+    File stderr = std.stdio.stderr;
+    string[string] env;
+    Config config = Config.none;
+    char[] workDir;
+}
+Pid spawnProcess(in char[][] args, SpawnParams params);
+```
+
+#### In-place struct initialization in the wild
+
+The following example of an interaction with Amazon Web Services
+is an example for how in-place struct initialization would help
+with the interaction with complex APIs:
+At the moment an complex configuration objects can't be easily constructed in-place,
+which gets pretty tedious for large objects:
+
+```d
+BucketOptions b1Options = {
+    bucket: "MyBucket1",
+    createBucketConfiguration: {
+        locationConstraint: BucketLocationConstraint.EU_CENTRAL_1
+    }
+};
+BucketOptions b1Options = {
+    bucket: "MyBucket2",
+    createBucketConfiguration: {
+        locationConstraint: BucketLocationConstraint.EU_CENTRAL_1
+    }
+};
+invoker.execute([
+	new CreateBucketCommand(client, b1Options),
+	new CreateBucketCommand(client, b2Options),
+])
+```
+
+With in-place struct initialization the configuration object can be constructed
+in-place and no intermediate states which are typically tedious to maintain are required:
+
+```d
+invoker.execute([
+	new CreateBucketCommand(client, BucketOptions(
+		bucket: "MyBucket1",
+		createBucketConfiguration: {
+			locationConstraint: BucketLocationConstraint.EU_CENTRAL_1
+		}
+	)),
+	new CreateBucketCommand(client, BucketOptions(
+		bucket: "MyBucket2",
+		createBucketConfiguration: {
+			locationConstraint: BucketLocationConstraint.EU_CENTRAL_1
+		}
+	))
+]);
+```
+
+Complex configuration objects is common day-to-day problem.
+Here's another configuration example from [Vulkan](https://github.com/ColonelThirtyTwo/dvulkan):
+
+```d
+VkImageCreateInfo imgInfo = {
+	imageType: VkImageType.VK_IMAGE_TYPE_2D,
+	format: VkFormat.VK_FORMAT_R8G8B8A8_UNORM,
+	extent: image.size,
+	mipLevels: image.mipLevels,
+	arrayLayers: image.layers,
+	samples: VkSampleCountFlagBits.VK_SAMPLE_COUNT_1_BIT,
+	tiling: VkImageTiling.VK_IMAGE_TILING_LINEAR,
+	usage: VkImageUsageFlagBits.VK_IMAGE_USAGE_TRANSFER_SRC_BIT | VkImageUsageFlagBits.VK_IMAGE_USAGE_SAMPLED_BIT,
+	sharingMode: VkSharingMode.VK_SHARING_MODE_EXCLUSIVE,
+	initialLayout: VkImageLayout.VK_IMAGE_LAYOUT_PREINITIALIZED,
+};
+enforceVK(vkCreateImage(device, &imgInfo, null, &image.img));
+```
+
+### User-defined attributes
+
+Currently on UDAs it's not possible to specify only specific argument nor arbitrary order.
+This DIP would allow this useful pattern:
+
+```d
+@S(c:42) void foo();
+```
+
+This is especially useful for serialization:
+
+```d
+struct CreateTableInput
+{
+	@FieldInfo({memberName: "TableName"})
+	TableName tableName;
+
+	@FieldInfo({memberName: "AttributeDefinitions", minLength: 1})
+	AttributeDefinitions attributeDefinitions;
+}
+```
+
```

Comment:
I think another great example would be to create a struct instance to be converted to JSON in an implementation of a HTTP API endpoint.